### PR TITLE
[#5170] optimizing folder rename operation

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Create a resource. Create a folder. Upload 500 or so files (more files is better for testing) to that folder. However, upload only 100 files at a time as there is a separate issue to fix large number of file upload.
2. Rename the folder. Verify the folder rename was successful without any timeout error.

### CODE REVIEW

File: hs_core/views/utils.py [hunk 1]
```diff
@@ -886,13 +886,24 @@ def rename_irods_file_or_folder_in_django(resource, src_name, tgt_name):
 
     except ObjectDoesNotExist:
         # src_name and tgt_name are folder names
-        res_file_objs = ResourceFile.list_folder(resource, src_name)
-
+        res_file_objs = ResourceFile.list_folder(resource=resource, folder=src_name)
+        resource_is_federated = resource.is_federated
         for fobj in res_file_objs:
-            src_path = fobj.storage_path
```
Comment (pkdash) [inline hs_core/views/utils.py:892]: This results in 1 db query per file.

File: hs_core/views/utils.py [hunk 2]
```diff
@@ -886,13 +886,24 @@ def rename_irods_file_or_folder_in_django(resource, src_name, tgt_name):
...
+            src_path = fobj.get_storage_path(resource=resource)
             # naively replace src_name with tgt_name
             new_path = src_path.replace(src_name, tgt_name, 1)
-            fobj.set_storage_path(new_path)
```
Comment (pkdash) [inline hs_core/views/utils.py:895]: This results in 3 db queries and 1 irods query per file.

**Review by devincowan — DISMISSED** [review]
Looks good thanks!
Couple of questions for you.
Also, we might be able to safely remove the duplicate `.save()` calls in [set_storage_path](https://github.com/hydroshare/hydroshare/blob/5170-optimize-folder-rename/hs_core/models.py#L3215-L3229)

File: hs_core/views/utils.py [hunk 3]
```diff
@@ -888,6 +888,7 @@ def rename_irods_file_or_folder_in_django(resource, src_name, tgt_name):
         # src_name and tgt_name are folder names
         res_file_objs = ResourceFile.list_folder(resource=resource, folder=src_name)
         resource_is_federated = resource.is_federated
+        batch_size = 10-000
```
Comment (devincowan) [inline hs_core/views/utils.py:891]: this is how `batch_size` is specified?
https://docs.djangoproject.com/en/3.2/ref/models/querysets/#bulk-update

Comment (devincowan) [reply hs_core/views/utils.py]: nice, sounds good

**Review by devincowan — APPROVED** [review]

**Pull request merged**